File: android/src/main/java/com/lightningkite/khrysalis/views/EntryPoint.shared.kt
package com.lightningkite.khrysalis.views

import com.lightningkite.khrysalis.observables.ObservableStack

interface EntryPoint: HasBackAction {
    fun handleDeepLink(schema: String, host: String, path: String, params: Map<String, String>){
        println("Empty handler; $schema://$host/$path/$params")
    }
    val mainStack: ObservableStack<ViewGenerator>? get() = null
}

File: ios/Khrysalis/com/lightningkite/khrysalis/views/EntryPoint.shared.swift
//Package: com.lightningkite.khrysalis.views
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay



public protocol EntryPoint {
    
    func handleDeepLink(schema: String, host: String, path: String, params: Dictionary<String, String>) -> Void
    func handleDeepLink(_ schema: String, _ host: String, _ path: String, _ params: Dictionary<String, String>) -> Void
    
    func onBackPressed() -> Bool
    
    var mainStack: ObservableStack<ViewGenerator>?  { get }
}

public extension EntryPoint {
    
    func handleDeepLink(schema: String, host: String, path: String, params: Dictionary<String, String>) -> Void {
        print("Empty handler; \(schema)://\(host)/\(path)/\(params)")
    }
    func handleDeepLink(_ schema: String, _ host: String, _ path: String, _ params: Dictionary<String, String>) -> Void {
        return handleDeepLink(schema: schema, host: host, path: path, params: params)
    }
    
    func onBackPressed() -> Bool {
        return false
    }
    public var mainStack: ObservableStack<ViewGenerator>?  {
        get {
            return nil
        }
    }
}
 
